velocity/tool/src/java/org/sakaiproject/cheftool/VelocityPortletPaneledAction.java

```diff
@@ -190,20 +190,30 @@ public static String titlePanelUpdateId(String toolId)
 
 	} // titlePanelUpdateId
 
+	// Backward compatible default for removing duplicates in addAlert
+	public static void addAlert(SessionState state, String message) {
+		
+		addAlert(state, message, false);
+	}
+
 	/**
 	 * Add another string to the alert message.
 	 * 
 	 * @param state
 	 *        The session state.
 	 * @param message
 	 *        The string to add.
+	 * @param removeDuplicates
+	 * 		  Remove duplicates from the alert
 	 */
-	public static void addAlert(SessionState state, String message)
+	public static void addAlert(SessionState state, String message, boolean removeDuplicates)
 	{
 		String soFar = (String) state.getAttribute(STATE_MESSAGE);
 		if (soFar != null)
 		{
-			soFar = soFar + "<br/>" + message;
+			if (!(removeDuplicates == true && soFar.contains(message))) {
```

Comment:
Easier code to read:

if (!removeDuplicates || !soFar.contains(message)) {

Reply:
Could reverse the null check and avoid the extra indentation:

if (soFar == null)
{
	soFar = message;
}
else if (!removeDuplicates || !soFar.contains(message))
{
	soFar += "<br/>" + message;
}

Reply:
Okay thanks changed it to this suggestion.